### DISCUSSION TITLE
How to write an intercept where I have to wait for 4 or 5 urls ?

### DISCUSSION THREAD

**Discussion by roopat** [cypress-io/cypress#16642] [General]
In my application , when I navigate to a new page , there are around 10+ GET calls happening which fetches 10+ different products on to the we page from the backend ...which means all the urls are same except the last parameter which is different for 10 different products ... i.e "productCatalogManagement/v4/productOfferingPrice/0CX-1X8QWZ" the last value is different for all 10+ different API calls . But I have to wait for all 10+ calls to proceed with my next operation ... I tried creating the intercept as 
cy.intercept ('**productCatalogManagement/v4/productOfferingPrice/**') ; 
But that is waiting for only few calls and then proceeding with the next operation which will fail as all the calls are not complete yet.
Can someone help me with this if they have a solution ?

**Comment by jennifer-shehane** [comment]
You may need to expand the glob you're using to listen match the route, depending on how the unique IDs look `productCatalogManagement/v4/productOfferingPrice/**` 

If you can get the matching to match all the routes, then you'd want to listen for 10 calls before continuing in Cypress, so you might want to use the `@alias.all` undocumented signature to wait for exactly 10 calls:

```js
cy.intercept(`productCatalogManagement/v4/productOfferingPrice/**`).as('getProduct')
cy.get('@getProduct.all').should('have.length', 10)
// next commands you want to run.
```

**Reply by roopat** [reply]
@jennifer-shehane I tried , but this is the error I am seeing 
"cy.wait() could not find a registered alias for: @getProduct.all.
Available aliases are: listPromotions, planConfig, productOfferingPrice." I am on 7.3.0 cypress version.Can you please help me out with this issue

If I use cy.get('@getProduct.all').should('have.length',10),  I am getting the below error 


expected [] to have a length of 20 but got 0

**Reply by jennifer-shehane** [reply]
Sorry, you're correct, it should be `cy.get('@getProduct.all')`. I updated my example above to reflect this. 

If it's not matching, then I would double check the glob to ensure it's written to match the full URL. You can try using `Cypress.minimatch` to check globs https://on.cypress.io/minimatch
